samples/cpp/intelligent_scissors.cpp

```diff
@@ -0,0 +1,201 @@
+                for(int j = -1; j <= 1; j++)
+                {
+                    int tx = p.x + i;
+                    int ty = p.y + j;
+                    if (tx < 0 || tx >= img.cols || ty < 0 || ty >= img.rows)
+                        continue;
+                    if (expand.at<uchar>(ty, tx) == 0)
+                    { 
+                        Point q = Point(tx, ty);
+                        float cost = cost_map.at<float>(p) + local_cost(p, q);
+                        if (processed.at<uchar>(q) == 1 && cost < cost_map.at<float>(q))
+                        {
+                            removed.at<uchar>(q) = 1;
+                        }
+                        if (processed.at<uchar>(q) == 0)
+                        {
+                            cost_map.at<float>(q) = cost;
+                            hit_map_x.at<int>(q)= p.x;
+                            hit_map_y.at<int>(q) = p.y;
+                            processed.at<uchar>(q) = 1;
+                            Pix val;
+                            val.cost = cost_map.at<float>(q);
+                            val.next_point = q;
+                            L.push(val);
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+void onMouse(int event, int x, int y, int flags, void *param)
+{
+    if (event == EVENT_LBUTTONDOWN)
+    {
+        end = Point(x, y);
+        if (!contours.back().empty())
+        {
+            for(int i = tmp_contour.size() - 1; i >= 0; i--)
+            {
+                contours.back().push_back(tmp_contour[i]);
+            }
+            tmp_contour.clear();
+        }
+        else
+        {
+            contours.back().push_back(end);
+        }
+        find_min_path(end);
+
+        img_render.copyTo(img_pre_render);
+        imshow("lasso", img_render);    
+    }
+    else if (event == EVENT_RBUTTONDOWN)
+    {
+        img_pre_render.copyTo(img_render);
+        drawContours(img_pre_render, contours, contours.size() - 1, Scalar(0,255,0), CV_FILLED);
+        addWeighted(img_pre_render, 0.3, img_render, 0.7, 0, img_render);
+        contours.resize(contours.size() + 1);
+        imshow("lasso", img_render);
+    }
+    else if (event == EVENT_MOUSEMOVE && !contours.back().empty())
+    {
+        tmp_contour.clear();
+        img_pre_render.copyTo(img_render);
+        Point val_point = Point(x, y);
+        while (val_point != end)
+        {
+            tmp_contour.push_back(val_point);
+            Point cur = Point(hit_map_x.at<int>(val_point), hit_map_y.at<int>(val_point));
+            line(img_render, val_point, cur, Scalar(255, 0, 0), 2);
+            val_point = cur;
+        }
+        imshow("lasso", img_render);
+    }
+}
+
+const char* keys =
+{
+    "{help h||}{@image |fruits.jpg|input image name}"
+};
+
+
+int main( int argc, const char** argv )
+{
+    CommandLineParser parser(argc, argv, keys);
+    std::string filename = parser.get<std::string>(0);
```

Comment:
Please process `-h (help)` flag and add some description to the sample. Check https://github.com/opencv/opencv/blob/471b40040a304e92bbe4a3a53e81b56dd00b2c1a/samples/dnn/object_detection.cpp#L116 for reference.

Reply:
We also need to add the instructions of how to use a sample - which buttons in which order.